packages/kbn-ftr-common-functional-services/services/saml_auth/serverless/auth_provider.ts

```diff
@@ -50,18 +50,35 @@ export class ServerlessAuthProvider implements AuthProvider {
     this.rolesDefinitionPath = resolve(SERVERLESS_ROLES_ROOT_PATH, this.projectType, 'roles.yml');
   }
 
-  getSupportedRoleDescriptors(): Record<string, unknown> {
-    return readRolesDescriptorsFromResource(this.rolesDefinitionPath) as Record<string, unknown>;
+  getSupportedRoleDescriptors() {
+    const roleDescriptors = new Map<string, any>(
+      Object.entries(
+        readRolesDescriptorsFromResource(this.rolesDefinitionPath) as Record<string, unknown>
+      )
+    );
+    if (this.projectType !== 'oblt') {
```

Comment:
Can you help me understand this?

Reply:
Custom roles are enabled on MKI only for their Elasticsearch and Elastic Security serverless projects, I replace it with a better condition and comment.

Reply:
Thanks for the explanation, lgtm